### PR TITLE
Recreate the entiry of DLC Restoration in LWAPI.

### PR DESCRIPTION
**Pull request by DeaTh-G**
Everything in this PR was mapped during the process of recreating https://github.com/DeaTh-G/slw-dlc-restoration/ in LWAPI.
The PR includes MANY API breaking changes and requires extensive reviewing.

### CODE REVIEW

File: Sonic2013/Hedgehog/Game/goc/goc_Transform.h [hunk 1]
```diff
@@ -37,6 +42,35 @@ namespace app::fnd
...
+				if (pParent)
+					pParent->RemoveChild(this);
+			
+				pParent = in_pParent;
+				if (in_pParent)
+					in_pParent->m_Children.push_back(this);
+			}
+		}
+
+		bool IsExistParent()
```
Comment (Sajidur78) [inline Sonic2013/Hedgehog/Game/goc/goc_Transform.h]: This should be const

File: Sonic2013/Hedgehog/Game/goc/hud/goc_Hud.h [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+				Option = in_option;
+				Size = in_size;
+				Unk2 = in_unk2;
+			}
+		};
+
+		int m_Unk1{};
+		csl::ut::ObjectMoveArray<SurfRide::ReferenceCount<SurfRide::Project>> rcProjects{};
+		csl::ut::MoveArray<game::HudLayerController*> m_LayerControllers{};
+		gfx::Renderable* pRenderable{}; // boost::intrusive_ptr<gfx::Renderable>
```
Comment (Sajidur78) [inline Sonic2013/Hedgehog/Game/goc/hud/goc_Hud.h]: Use `app::ut::RefPtr` in place of `boost::intrusive_ptr` and leave the boost type as a comment. RefPtr and intrusive_ptr serve the same functionality.

**Review by Sajidur78 — CHANGES_REQUESTED** [review]

File: Sonic2013/Hedgehog/Game/goc/hud/goc_Hud.h (hunk 2 above)
Comment (DeaTh-G) [reply Sonic2013/Hedgehog/Game/goc/hud/goc_Hud.h]: Changed in bec8118.

File: Sonic2013/Hedgehog/Game/goc/goc_Transform.h (hunk 1 above)
Comment (DeaTh-G) [reply Sonic2013/Hedgehog/Game/goc/goc_Transform.h]: Changed in fa70526.

**Review by Sajidur78 — APPROVED** [review]
LGTM

**Pull request merged**